### PR TITLE
Greg/common make unique

### PR DESCRIPTION
**Pull request by gregory38**
Question do we also do it on CDVDgiga and GSdx ?

### CODE REVIEW

**Comment by gregory38** [comment]
Hum is C++14 enabled on Visual Studio ? Or does it export std::make_unique even on C++11 build ?

**Comment by turtleli** [comment]
make_unique has been available since VS2013.

I think both VS2015 and VS2017 use C++14.

**Comment by gregory38** [comment]
I'm not sure it is a good idea to enable C++14 construct. I mean that C++11 code will compile all compliers. C++14 code might depends on the compiler.

**Comment by turtleli** [comment]
There isn't an option to compile as C++11 with VS. Right now it's either C++14, or C++ latest (follows latest drafts).

File: common/include/Utilities/MakeUnique.h [hunk 1]
```diff
@@ -23,7 +23,7 @@
 // Instead just provide the std++ implementation when only C++11 is enabled
 // File could be dropped when we switch to C++14
 
-#if __cplusplus <= 201103L
+#if __cplusplus <= 201103L && !defined(_WIN32)
```
Comment (mirh) [inline common/include/Utilities/MakeUnique.h]: `_MSC_VER` feels more orthodox 👅

Comment (gregory38) [inline common/include/Utilities/MakeUnique.h]: You're right but I fail to find the define name. I will update it.

**Comment by gregory38** [comment]
```
There isn't an option to compile as C++11 with VS. Right now it's either C++14, or C++ latest (follows latest drafts).
```
Oh. Ok. It isn't nice but we will manage.

**Comment by gregory38** [comment]
https://blogs.msdn.microsoft.com/vcblog/2016/06/07/standards-version-switches-in-the-compiler/
```
Note that there is no plan to add a C++11 switch.

There are some post-C++14 features already implemented in previous releases of the Visual C++ compiler. To avoid breaking changes for users who have already taken dependencies on these features we’ll leave them under the /std:c++14 switch. They include the following features, all of which have already shipped in Visual Studio 2015 Update 2:

    New rules for auto with braced-init-lists
    typename in template template-parameters
    Removing trigraphs
    Attributes for namespaces and enumerators
    u8 character literals
```
So the switch enables some C++17 construct. And I don't think they support all C++14... 

Nevertheless, it would be nice to enable the C++14 switch.

**Comment by mirh** [comment]
> So the switch enables some C++17 construct. And I don't think they support all C++14 (in VS2015)...

https://docs.microsoft.com/en-us/cpp/visual-cpp-language-conformance
Afaiu they are just missing a couple of C++14 features in VS2015. 

[In other news](https://blogs.msdn.microsoft.com/vcblog/2017/04/17/cmake-support-in-visual-studio-whats-new-in-2017-15-2-update/), seems like they are pretty quick with updating their CMake support.

**Comment by gregory38** [comment]
Well C++14 isn't that big, so a couple is nearly half ;) The point is to have the same feature set on all compilers. I found it easier this way but I'm maybe oldschool.

Hopefully someone will plug CMake/VS properly on PCSX2.

File: common/include/Utilities/MakeUnique.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2010  PCSX2 Dev Team
```
Comment (turtleli) [inline common/include/Utilities/MakeUnique.h]: Wrong copyright date?

Comment (gregory38) [inline common/include/Utilities/MakeUnique.h]: Done. thank you.

**Pull request merged**